src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Runtimes/MissingSdkRuntimeDetector.cs

```diff
@@ -48,32 +53,36 @@ protected override Task DisposeCoreAsync(bool initialized)
 
         protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
         {
-            // Note we don't use the ISafeProjectGuidService here because it is generally *not*
-            // safe to use within IProjectDynamicLoadComponent.LoadAsync.
```

Comment:
Why remove this comment? It seems like it could prevent errors in future refactoring.

Reply:
Should this comment be at this place ? sounds like this warning is more general.

Reply:
This was added recently by @tmeschter. It aims to prevent us trying to change this in future, as described.

Reply:
added this comments back